_inc/client/my-plan/checklist-progress-card/progress-bar/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { PureComponent } from 'react';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import ScreenReaderText from './screen-reader-text';
+
+export default class ProgressBar extends PureComponent {
+	static defaultProps = {
+		total: 100,
+		compact: false,
+		isPulsing: false,
+		canGoBackwards: false,
+	};
+
+	static propTypes = {
+		value: PropTypes.number.isRequired,
+		total: PropTypes.number,
+		color: PropTypes.string,
+		title: PropTypes.string,
+		compact: PropTypes.bool,
+		className: PropTypes.string,
+		isPulsing: PropTypes.bool,
+		canGoBackwards: PropTypes.bool,
+	};
+
+	static getDerivedStateFromProps( props, state ) {
+		return {
+			allTimeMax: Math.max( state.allTimeMax, props.value ),
+		};
+	}
+
+	state = {
+		allTimeMax: this.props.value,
+	};
+
+	getCompletionPercentage() {
+		const percentage = Math.ceil(
+			( ( this.props.canGoBackwards ? this.props.value : this.state.allTimeMax ) /
+				this.props.total ) *
+				100
+		);
+
+		// The percentage should not be allowed to be more than 100
+		return Math.min( percentage, 100 );
+	}
+
+	renderBar() {
+		const title = this.props.title ? (
+			<ScreenReaderText>{ this.props.title }</ScreenReaderText>
+		) : null;
+
+		const styles = { width: this.getCompletionPercentage() + '%' };
+		if ( this.props.color ) {
+			styles.backgroundColor = this.props.color;
+		}
```

Comment:
What do you think if we'd drop this inline style prop from this component? It's used just once in Calypso and I'd like to remove it there, too.

Reply:
A fair amount of the functionality is unused. Do you think I should strip this down to be a custom progress bar just for our case?

I'm not sure whether it's best to make it easy to replace with the Calypso version or specialized for this use.

Reply:
True, prolly not worth it. 👍